test/integration/examples/apiserver_test.go

```diff
@@ -47,41 +61,266 @@ var groupVersionForDiscovery = metav1.GroupVersionForDiscovery{
 	Version:      groupVersion.Version,
 }
 
-func TestRunServer(t *testing.T) {
-	masterConfig := framework.NewIntegrationTestMasterConfig()
-	_, s := framework.RunAMaster(masterConfig)
-	defer s.Close()
+// findAvailableBindAddress returns a bind address on 127.0.0.1 with a free port in the low-high range.
+// If lowPort is 0, an ephemeral port is allocated.
+func findAvailableBindAddress(lowPort, highPort int) (int, error) {
```

Comment:
(no action required) Why is it desirable to specify a range and iterate through it? The strategy employed by ``httptest`` package's ``newLocalListener`` method seems simpler - listen on port 0 so that a port is automatically allocated by the system.

``httptest``'s approach also has the advantage of keeping the port bound.  Since there is no guarantee that the port chosen by this method will still be available after the listener is closed, consider adding error handling around server start to prevent flakes related to the port being taken by another process after it has been chosen.

Is there a reason to prefer the approach here vs revising how a server is started to take advantage of ``httptest``?  I have a similar need for dynamic port allocation in running the federation api in testing.